src/reducer.js

```diff
@@ -0,0 +1,33 @@
+const initialState = {
+  newId: 100,
+  taskTitle: '',
+  tasks: [],
+};
+
+export default function reducer(state = initialState, action) {
+  const actionsTable = {
+    updateTaskTitle: () => ({
+      ...state,
+      taskTitle: action.payload.taskTitle,
+    }),
+
+    addTask: () => {
+      if (!state.taskTitle) {
+        return state;
+      }
+
+      return {
+        newId: state.newId + 1,
+        taskTitle: '',
+        tasks: [...state.tasks, { id: state.newId, title: state.taskTitle }],
+      };
+    },
+
+    deleteTask: () => ({
+      ...state,
+      tasks: state.tasks.filter((task) => task.id !== action.payload.id),
+    }),
+  };
+
+  return actionsTable[(action?.type)] ? actionsTable[(action?.type)]() : state;
```

Comment:
```suggestion
  return actionsTable[action?.type] ? actionsTable[action?.type]() : state;
```
소괄호로 감싸주지 않아도 되지 않나요?

그리고 action에 type이 없는 경우를 위해 reducer 함수에서 action에 default value로 초기값을 넣어줘도 될 것 같네요 ㅎㅎ

Reply:
말씀대로 초기값을 넣으니
action이 없어 발생하는 참조에러에 대한 처리도 필요 없어지는군요.